Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -386,11 +420,20 @@ bool AtBreakpoint()
 
 void RunGpu()
 {
+	SCPFifoStruct &fifo = CommandProcessor::fifo;
 	if (SConfig::GetInstance().m_LocalCoreStartupParameter.bCPUThread &&
 	    !g_use_deterministic_gpu_thread)
+	{
+		if (!s_gpu_is_running.load())
+		{
+			std::unique_lock<std::mutex> lock(s_fifo_mutex);
```

Comment:
I don't think so. Line 374 is just for the fastpath to check twice. For the race condition, line 384 "[] () { return s_gpu_is_running.load(); }" is executed within the lock before waiting the first time.